src/components/RewardsSection/index.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import useEmblaCarousel from 'embla-carousel-react'
+import styles from './RewardsSection.module.scss'
+import RewardsCard, { RewardsCardProps } from '@/components/RewardsCard'
+import ArrowCircularButton from '../ArrowCircularButton'
+
+export interface RewardsSectionProps {
+  rewards: RewardsCardProps[]
+  Link: React.ComponentType<{ children: React.ReactNode; href: string }>
+  isLoading?: boolean
```

Comment:
this should be able to be set true and the section then shows in a loading skeleton state. 

i.e. we should prepoluate the section with a few dummy cards and have those in a loading state rather than overriding `rewards`.

Reply:
yeah, this was what I imagined. Will add those cards.

Reply:
done

Reply:
So, in the end adding the dummy cards here adds a lot of edge cases and one of them I could not fix on the store:
Ember Sword: there is a quest for it, but no rewards. This makes the section shows but loading forever.
I think might be safer to use a similar approach you did before, but I will add the dummy data on another file.

Reply:
Actually, even moving the dummy data to the store embers sword still loads forever because this is true:
`props.quests.length > 0 `.

Reply:
I think that since this is an edge case, I will add a timeout for the isloading state and then we will have layout shift but just for a couple of games. What do you think@